pkg/mic/mic.go

```diff
@@ -1099,6 +1099,7 @@ func (c *Client) updateUserMSI(newAssignedIDs map[string]aadpodid.AzureAssignedI
 		}
 
 		for _, createID := range createOrUpdateList {
+			createID := createID
```

Comment:
Let's add a comment here on why this is being done. Do we need a new instance of the same variable? I know this would be the case when the variables are passed to anonymous functions.

Reply:
Thank you for the detailed description. I wanted to have this documented in the PR for future context 😄

Option 1 is an idempotent way to do it. So we'll stick with that.